pom.xml

```diff
@@ -474,7 +482,7 @@
                         <dependency>
                             <groupId>com.puppycrawl.tools</groupId>
                             <artifactId>checkstyle</artifactId>
-                            <version>8.0</version>
+                            <version>8.11</version>
```

Comment:
We may want to copy over the latest checkstyle ruleset xml file?  Or maybe you did already haven't read that far into this PR yet.

Reply:
I did, I also kept a few differences and added the unused import filter.